frontend/js/dirViewer.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+import Component from './component.js';
+
+export default class DirViewer extends Component{
+    constructor(options) {
+        super(options);
+        this._selectedItem = null;
+
+        this.on('click',this._onClick.bind(this));
+        this.on('dblclick',this._onDblClick.bind(this));
+        this.on('mousedown', DirViewer.preventDefaultSelection);
+        this.on('selectstart',DirViewer.preventDefaultSelection);
+    }
+
+    static preventDefaultSelection(event) {
+        event.preventDefault();
+    }
+
+    static _definePath(target) {
+        let path = '';
+        let dir = target;
+
+        while(dir) {
+            path = '/' + dir.textContent + path;
+            dir = dir.parentNode.parentNode.previousElementSibling;
+        }
+
+        return path;
+    }
+
+    get selectedItem() {
+        return this._selectedItem;
+    }
+
+    _onClick(event) {
+
+        let target = event.target.closest('span');
+
+        if(!target) {
+            return;
+        }
+
+        if (target.nextElementSibling) {
+            DirViewer.toggleElement(target.nextElementSibling);
+        }
+
+
+        if (this.selectedItem) {
+            DirViewer.toggleFocus(this.selectedItem);
+        }
+
+        this._selectedItem = target;
+        DirViewer.toggleFocus(target);
+
+        this._trigger('itemWasSelected', {path: DirViewer._definePath(target)});
+
+    }
+
+    _onDblClick(event) {
+
```

Comment:
this functionality is not obvious, it is better to write few words about it at the top of page. I didnt know about it until I've see the this line)